src/rules/fileNameCasingRule.ts

```diff
@@ -43,18 +101,46 @@ export class Rule extends Lint.Rules.AbstractRule {
             * \`${Casing.SnakeCase}\`: File names must be snake-cased: \`file_name.ts\`.`,
         options: {
             type: "array",
-            items: [
-                {
-                    type: "string",
-                    enum: [Casing.CamelCase, Casing.PascalCase, Casing.KebabCase, Casing.SnakeCase],
-                },
-            ],
+            items: {
+                anyOf: [
+                    {
+                        type: "array",
+                        items: [
+                            {
+                                type: "string",
+                                enum: rules,
+                            },
+                        ],
+                    },
+                    {
+                        type: "object",
+                        /**
+                         * TODO: Add valid validation
+                         * We do not care about the key but the value most be
+                         * an string of rules enum.
+                         *
+                         * Example:
+                         *      properties: valuesOf({
+                         *          type: "string",
+                         *          enum: rules
+                         *      })
+                         */
+                    },
+                ],
+            },
         },
         optionExamples: [
             [true, Casing.CamelCase],
             [true, Casing.PascalCase],
             [true, Casing.KebabCase],
             [true, Casing.SnakeCase],
+            [
+                true,
+                {
+                    ".tsx": "pascal-case",
+                    ".ts": "camel-case",
```

Comment:
Nit: `"tsx": Casing.SnakeCase` and `".ts": Casing.CamelCase` please, per the convention in the other examples.

Reply:
I don't understand what you mean by `per the convention in the other examples`.

The other examples are about a single value, this is showing the example for a complex setup.